### PR TITLE
ExpectedConditions utility class added

### PR DESCRIPTION
**Pull request by Docik**

### CODE REVIEW

**Review by tummima — APPROVED** [review]

File: src/main/groovy/io/cify/framework/core/CifyExpectedConditions.groovy [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package io.cify.framework.core
+
+import org.openqa.selenium.Point
+import org.openqa.selenium.StaleElementReferenceException
+import org.openqa.selenium.WebDriver
+import org.openqa.selenium.WebElement
+import org.openqa.selenium.support.ui.ExpectedCondition
+
+class CifyExpectedConditions {
```
Comment (wave911) [inline src/main/groovy/io/cify/framework/core/CifyExpectedConditions.groovy]: Does it make sense rather extend "org.openqa.selenium.support.ui.ExpectedCondition" ?

File: src/main/groovy/io/cify/framework/core/CifyExpectedConditions.groovy [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                Point currentLocation = element.location
+                try {
+                    if (currentLocation.x == lastLocation.x && currentLocation.y == lastLocation.y) {
+                        lastLocation = new Point(0, 0)
+                        return true
+                    }
+                    lastLocation.x = currentLocation.x
+                    lastLocation.y = currentLocation.y
+                    return false
+                } catch (StaleElementReferenceException ignored) {
```
Comment (wave911) [inline src/main/groovy/io/cify/framework/core/CifyExpectedConditions.groovy]: If another expected occurs - what will be returned from the method?

File: src/main/groovy/io/cify/framework/core/CifyExpectedConditions.groovy (hunk 1 above)
Comment (tummima) [reply src/main/groovy/io/cify/framework/core/CifyExpectedConditions.groovy]: org.openqa.selenium.support.ui.ExpectedCondition works a bit differently and often is not as reliable

Comment (wave911) [reply src/main/groovy/io/cify/framework/core/CifyExpectedConditions.groovy]: @tummima could you please elaborate a bit? What do you under "not reliable"?

File: src/main/groovy/io/cify/framework/core/CifyExpectedConditions.groovy (hunk 2 above)
Comment (tummima) [reply src/main/groovy/io/cify/framework/core/CifyExpectedConditions.groovy]: other exceptions will not be caught and the app would crash

File: src/main/groovy/io/cify/framework/core/CifyExpectedConditions.groovy (hunk 1 above)
Comment (tummima) [reply src/main/groovy/io/cify/framework/core/CifyExpectedConditions.groovy]: First org.openqa.selenium.support.ui.ExpectedCondition can be applied only to certain cases, you can try this in test code, when you import the library, it is not as flexible, thus custom conditions need to be made. Second "works a bit differently" means that they are made for more common cases and when using them we should fully understand what they do, thus when implementing it is easy to make the code "not reliable".

File: src/main/groovy/io/cify/framework/core/CifyExpectedConditions.groovy (hunk 2 above)
Comment (wave911) [reply src/main/groovy/io/cify/framework/core/CifyExpectedConditions.groovy]: It might crash in some other place. What might raise an exception here?

Comment (tummima) [reply src/main/groovy/io/cify/framework/core/CifyExpectedConditions.groovy]: unknown issue with unknown exception

File: src/main/groovy/io/cify/framework/core/CifyExpectedConditions.groovy (hunk 1 above)
Comment (Docik) [reply src/main/groovy/io/cify/framework/core/CifyExpectedConditions.groovy]: > Does it make sense rather extend "org.openqa.selenium.support.ui.ExpectedCondition" ?

It does, but "org.openqa.selenium.support.ui.ExpectedCondition" has private constructor=(

File: src/main/groovy/io/cify/framework/core/CifyExpectedConditions.groovy (hunk 2 above)
Comment (Docik) [reply src/main/groovy/io/cify/framework/core/CifyExpectedConditions.groovy]: I don't see any other exceptions to be thrown here...

**Review by wave911 — APPROVED** [review]

**Pull request merged**